include/generic_sort.h

```diff
@@ -41,7 +45,11 @@ std::string to_smtlib(SortKind);
 class GenericSort : public AbsSort
 {
  public:
-  GenericSort(SortKind sk); 
+  GenericSort(SortKind sk);
+  // This second constructor is only for datatype sorts
+  GenericSort(SortKind sk, std::string sort_name);
```

Comment:
In the line above, please remove the word "second" -- I am not sure if it will always stay the second constructor, but I am sure that if it doesn't we will forget to update the comment :)